File: src/front-end-nextjs/hob-dashboard/lib/types/customer.ts
export interface Customer {
  customerId: string;
  name: string;
  email: string;
  phone?: string;
  createdAt: string;
  updatedAt: string;
  orders: OrderSummary[];
}

export interface OrderSummary {
  orderId: string;
  orderDate: string;
  totalAmount: number;
  status: string;
}

export interface CreateCustomerRequest {
  name: string;
  email: string;
  phone?: string;
}

export interface UpdateCustomerRequest {
  name: string;
  email: string;
  phone?: string;
}

File: src/back-end-dotnet/HOB.Data/Entities/Customer.cs
namespace HOB.Data.Entities;

public class Customer
{
    public Guid CustomerId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation property
    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}
